### PR TITLE
Add a section to the development manual on the structure of the runtime libraries

### PR DESCRIPTION
**Pull request by rsmmr**

### CODE REVIEW

File: doc/development/architecture.rst [hunk 1]
```diff
@@ -4,4 +4,113 @@
...
+    Private Spicy-side library functionality that the HILTI code
+    coming out of Spicy compilation can ``import`` (e.g., functions to
+    access parsing input, such as ``spicy_rt::waitForInput()``;
+    HILTI-side type definitions for Spicy-specific types, such as for
+    a ``sink``). This functionality is declared in
+    ``spicy/lib/spicy_rt.spicy`` and implemented in C++ in
+    ``libspicy-rt.a``.
+
+Category 4
+    Public library functionality for C++ host applications to
```
Comment (bbannier) [inline doc/development/architecture.rst:59]: Can we change to ordering of these? With that 1-3 would all be public functionality and 4-5 private.

File: doc/development/architecture.rst [hunk 2]
```diff
@@ -4,4 +4,113 @@
...
+        categories.
+
+.. rubric:: What goes where?
+
+Think of Category 1 as the "Spicy standard library": functionality for
+user-side Spicy code to leverage.
+
+Category 2 is the same for HILTI, except that the universe of HILTI
+users remains extremely small right now (it’s just Spicy and people
+writing tests :-).
```
Comment (bbannier) [inline doc/development/architecture.rst:97]: Are we missing the closing parenthesis here?

File: doc/development/architecture.rst (hunk 1 above)
Comment (rsmmr) [reply doc/development/architecture.rst:59]: Ok, I was debating about that, will do.

Comment (rsmmr) [reply doc/development/architecture.rst]: Done

**Pull request merged**